selfservice/strategy/saml/strategy/test/strategy_test.go

```diff
@@ -1,23 +1,242 @@
-package strategy_test
+package saml_test
 
 import (
 	"bytes"
 	"context"
 	"encoding/json"
+	"github.com/gofrs/uuid"
+	"github.com/ory/kratos/driver/config"
+	"github.com/ory/kratos/internal"
+	"github.com/ory/kratos/internal/testhelpers"
+	"github.com/ory/kratos/selfservice/flow"
+	"github.com/ory/kratos/selfservice/flow/login"
+	"github.com/ory/kratos/selfservice/strategy/saml"
+	"github.com/ory/kratos/ui/container"
+	"github.com/ory/kratos/x"
+	"github.com/ory/x/assertx"
+	"github.com/ory/x/urlx"
+	"github.com/stretchr/testify/assert"
+	"github.com/tidwall/gjson"
+	"golang.org/x/net/html"
+	"io"
+	"net/http"
+	"net/http/cookiejar"
+	"net/http/httptest"
+	"net/url"
 	"regexp"
+	"strings"
 	"testing"
+	"time"
 
 	"github.com/ory/kratos/identity"
 	samlhandler "github.com/ory/kratos/selfservice/flow/saml"
-	helpertest "github.com/ory/kratos/selfservice/flow/saml/helpertest"
+	"github.com/ory/kratos/selfservice/flow/saml/helpertest"
 	samlstrategy "github.com/ory/kratos/selfservice/strategy/saml/strategy"
 	"github.com/stretchr/testify/require"
-
-	"gotest.tools/assert"
-	gotest "gotest.tools/assert"
-	is "gotest.tools/assert/cmp"
 )
 
+const debugRedirects = false
+
+func TestStrategy(t *testing.T) {
```

Comment:
Generally speaking, there could be more comments in this PR. To detail what test cases do, what are the expected outcome ... to have the main lines of what the code actually do.